shapeshift/web
# chore: update SupportedWalletInfo to support mobile app

## Description

Add a configuration option to specify if the wallet should be shown in the mobile app and/or mobile browser

Change `mobileEnabled` wallet configuration flag to `supportsMobile` with 4 possible values (including `undefined`)

This flag will be leveraged in the future to show a different set available wallets in the mobile app vs. mobile browser. Since the "ShapeShift" wallet actually works in the app, but without the security of the phone's secure storage, the plan is to show a different "ShapeShift Mobile" wallet type instead where the seed phrase is stored in secure storage.

## Notice




## Pull Request Type


## Issue (if applicable)

Related to https://github.com/shapeshift/mobile-app/issues/11

## Risk
Very low.

## Testing
1. Load the app
2. Click Connect Wallet and verify that all wallets are there
3. In Dev Tools / Elements, click the Mobile button to switch to mobile view
4. RELOAD the app
5. Click Connect Wallet and verify that the same list of mobile wallets is there

## Screenshots (if applicable)
![image](https://user-images.githubusercontent.com/1958266/184025571-6ab78600-2677-47fa-a855-1fc1ffa77fd0.png)
![image](https://user-images.githubusercontent.com/1958266/184025612-ab04e70b-7696-45c4-9d5c-7d76685124ce.png)
